app/clients/zendesk.py

```diff
@@ -0,0 +1,96 @@
+                    f"- Purpose: {self.contact.main_use_case}",
+                    f"- Notification types: {self.contact.notification_types}",
+                    f"- Expected monthly volume: {self.contact.expected_volume}",
+                    "---",
+                    self.contact.service_url,
+                ]
+            )
+        elif self.contact.is_branding_request():
+            message = "<br>".join(
+                [
+                    f"A new logo has been uploaded by {self.contact.name} ({self.contact.email_address}) for the following service:",
+                    f"- Service id: {self.contact.service_id}",
+                    f"- Service name: {self.contact.service_name}",
+                    f"- Logo filename: {self.contact.branding_url}",
+                    "<hr>",
+                    f"Un nouveau logo a été téléchargé par {self.contact.name} ({self.contact.email_address}) pour le service suivant :",
+                    f"- Identifiant du service : {self.contact.service_id}",
+                    f"- Nom du service : {self.contact.service_name}",
+                    f"- Nom du fichier du logo : {self.contact.branding_url}",
+                ]
+            )
+
+        if len(self.contact.user_profile):
+            message += f"<br><br>---<br><br> {self.contact.user_profile}"
+
+        return message
+
+    # Update for Zendesk API Ticket format
+    # read docs: https://developer.zendesk.com/rest_api/docs/core/tickets#create-ticket
+    def _generate_ticket(self) -> Dict[str, Union[str, int, List[str]]]:
+        product_id = current_app.config[" "]
+        if not product_id:
+            raise NotImplementedError
+
+        return {
+            "product_id": int(product_id),
+            "subject": self.contact.friendly_support_type,
+            "comment": self._generate_description(),
+            "email": self.contact.email_address,
+            "priority": 1,
+            "status": 2,
+            "tags": self.contact.tags,
+        }
+
+    def send_ticket(self) -> int:
+        try:
+            api_url = current_app.config["ZENDESK_API_URL"]  # Zendesk API URL
+            if not api_url:
+                raise NotImplementedError
+
+            # The API and field definitions are defined here:
+            # https://developer.zendesk.com/rest_api/docs/support/tickets
+            response = requests.post(
+                urljoin(api_url, "/api/v2/tickets"),
+                json=self._generate_ticket(),
+                auth=HTTPBasicAuth(current_app.config["ZENDESK_API_KEY"], "x"),
+                timeout=5,
+            )
+            response.raise_for_status()
+
+            return response.status_code
+        except requests.RequestException as e:
+            content = json.loads(response.content)
+            current_app.logger.warning(f"Failed to create Zendesk ticket: {content['errors']}")
+            raise e
```

Comment:
If we raise here then we logging it previously is not necessary. I would replicate what the `ZenDeskSell` class does and delegate responsibility of exception handling to the caller entirely, else we will end up with duplicate log entries.

I see that you followed what `freshdesk.py` did but `zendesk_sell.py` would have better practices IMHO.

Reply:
changed so that the error handling is done in one place (caller)